### DISCUSSION TITLE
数学公式重复出现

### DISCUSSION THREAD

**Discussion by lghnb** [lobehub/lobehub#3699] [General Question] [answered]
### 📦 Environment

Vercel

### 📌 Version

1.14.10

### 💻 Operating System

Windows

### 🌐 Browser

Firefox

### 🐛 Bug Description

数学公式重复出现，第一次公式渲染无误，随后出现相同的公式，而且渲染不正常

### 🚦 Expected Behavior

仅显示正常渲染的公式

### 📝 Additional Information

![图片](https://github.com/user-attachments/assets/81d2f197-e592-4d58-ae0f-0e336d6821cb)

[a comment by an automated account is not shown]

**Comment by lghnb** [comment]
切换源码显示
![图片](https://github.com/user-attachments/assets/9b13d2c0-71d6-4528-999b-04b5111c03fa)

**Comment by zhuozhiyongde** [comment]
我这边是正常的，另外你提交源码的时候可以用 Markdown 包裹起来，方便复制。

![image](https://github.com/user-attachments/assets/47bb8a32-3554-443d-b7a2-b112b47979cf)

这个问题一般见于 KaTeX 的 CSS 样式表没能被正常加载，请检查你的域名是否可以正常访问 CDN 资源，打开控制台、网络看看有无报错

[a comment by an automated account is not shown]

**Comment by sxjeru** [comment, marked as answer]
你大概也在用 .top 域名吧，cdn 把这个顶级域名拉黑了。
希望以后 lobechat 能让用户自选 cdn 。

![image](https://github.com/user-attachments/assets/b4af490c-0d74-4b43-b26c-286759101cea)

**Reply by arvinxx** [reply]
前段时间加了一个环境变量 `CDN_USE_GLOBAL=1`，可以将 cdn 切到 unpkg。

但我个人仍然强烈建议，不要使用 `.top` 域名。

**Reply by enor2017** [reply]
> 前段时间加了一个环境变量 `CDN_USE_GLOBAL=1`，可以将 cdn 切到 unpkg。
> 
> 但我个人仍然强烈建议，不要使用 `.top` 域名。

感谢回复，我这里尝试了一下`CDN_USE_GLOBAL=1`之后katex还是会请求相同的网址，这里是不是跟`aliyun`还是`unpkg` CDN无关？

**Comment by sxjeru** [comment]
作为临时措施，可以用油猴脚本在网页中插入其他来源的 css 。

```js
(function() {
    'use strict';

    var link = document.createElement('link');
    link.rel = 'stylesheet';
    link.type = 'text/css';

    link.href = 'https://lf9-cdn-tos.bytecdntp.com/cdn/expire-1-M/KaTeX/0.15.2/katex.min.css';

    document.head.appendChild(link);
})();
```

[a comment by an automated account is not shown]

**Comment by enor2017** [comment]
Hi all, thank you for the discussion and the temporary solutions. However, I believe a client-side solution by inserting CSS using plugins is not a proper solution -- I cannot expect clients to do this every time manually. May I know if there is any workaround to solve this from the server side? I am using docker for deployment, so I understand that might be more difficult.

**Comment by sxjeru** [comment]
@enor2017 Personally, I think it's a pity that unless Lobechat makes a change, it will be difficult to fix it from our own server side. 
Changing a domain seems to be another permanent solution.

**Reply by enor2017** [reply]
I see, thank you so much for your prompt reply!
